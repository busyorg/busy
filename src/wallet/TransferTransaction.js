import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage, FormattedRelative, FormattedDate, FormattedTime } from 'react-intl';
import { Tooltip } from 'antd';
import Avatar from '../components/Avatar';

const TransferTransaction = ({ to, memo, amount, timestamp }) => (
  <div className="UserWalletTransactions__transaction">
    <div className="UserWalletTransactions__avatar">
      <Avatar username={to} size={40} />
    </div>
    <div className="UserWalletTransactions__content">
      <div className="UserWalletTransactions__content-recipient">
        <FormattedMessage
          id="transferred_to"
          defaultMessage="Transferred to {username}"
          values={{
            username: <Link to={`/@${to}`}>{to}</Link>,
          }}
        />
        <span className="UserWalletTransactions__transfer">
          {'- '}{amount}
        </span>
      </div>
      <span className="UserWalletTransactions__timestamp">
        <Tooltip
          title={
            <span>
              <FormattedDate value={`${timestamp}Z`} />{' '}
              <FormattedTime value={`${timestamp}Z`} />
            </span>
          }
        >
          <span><FormattedRelative value={`${timestamp}Z`} /></span>
        </Tooltip>
      </span>
      <span className="UserWalletTransactions__memo">
        {memo}
      </span>
    </div>
  </div>
);

TransferTransaction.propTypes = {
  to: PropTypes.string,
  memo: PropTypes.string,
  amount: PropTypes.element,
  timestamp: PropTypes.string,
};

TransferTransaction.defaultProps = {
  to: '',
  memo: '',
  amount: <span />,
  timestamp: '',
};

export default TransferTransaction;
