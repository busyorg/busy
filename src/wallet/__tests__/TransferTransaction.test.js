import React from 'react';
import { shallow } from 'enzyme';
import TransferTransaction from '../TransferTransaction';

describe('(Component) TransferTransaction', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        to: 'hellosteem',
        memo: 'Test Transfer Transaction',
        amount: <span>{'0 STEEM'}</span>,
        timestamp: '0',
      };
      const wrapper = shallow(<TransferTransaction {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
