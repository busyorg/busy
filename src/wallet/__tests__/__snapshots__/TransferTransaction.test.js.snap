// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TransferTransaction with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="UserWalletTransactions__transaction"
  >
    <div
      className="UserWalletTransactions__avatar"
    >
      <Avatar
        size={40}
        username="hellosteem"
      />
    </div>
    <div
      className="UserWalletTransactions__content"
    >
      <div
        className="UserWalletTransactions__content-recipient"
      >
        <FormattedMessage
          defaultMessage="Transferred to {username}"
          id="transferred_to"
          values={
            Object {
              "username": <Link
                replace={false}
                to="/@hellosteem"
              >
                hellosteem
              </Link>,
            }
          }
        />
        <span
          className="UserWalletTransactions__transfer"
        >
          - 
          100 STEEM
        </span>
      </div>
      <span
        className="UserWalletTransactions__timestamp"
      >
        <FormattedRelative
          updateInterval={10000}
          value="0Z"
        />
      </span>
      <span
        className="UserWalletTransactions__memo"
      >
        Test Transfer Transaction
      </span>
    </div>
  </div>,
  "nodes": Array [
    <div
      className="UserWalletTransactions__transaction"
    >
      <div
        className="UserWalletTransactions__avatar"
      >
        <Avatar
          size={40}
          username="hellosteem"
        />
      </div>
      <div
        className="UserWalletTransactions__content"
      >
        <div
          className="UserWalletTransactions__content-recipient"
        >
          <FormattedMessage
            defaultMessage="Transferred to {username}"
            id="transferred_to"
            values={
              Object {
                "username": <Link
                  replace={false}
                  to="/@hellosteem"
                >
                  hellosteem
                </Link>,
              }
            }
          />
          <span
            className="UserWalletTransactions__transfer"
          >
            - 
            100 STEEM
          </span>
        </div>
        <span
          className="UserWalletTransactions__timestamp"
        >
          <FormattedRelative
            updateInterval={10000}
            value="0Z"
          />
        </span>
        <span
          className="UserWalletTransactions__memo"
        >
          Test Transfer Transaction
        </span>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <TransferTransaction
        amount="100 STEEM"
        memo="Test Transfer Transaction"
        timestamp="0"
        to="hellosteem"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "amount": "100 STEEM",
          "memo": "Test Transfer Transaction",
          "timestamp": "0",
          "to": "hellosteem",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="UserWalletTransactions__transaction"
        >
          <div
            className="UserWalletTransactions__avatar"
          >
            <Avatar
              size={40}
              username="hellosteem"
            />
          </div>
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <FormattedMessage
                defaultMessage="Transferred to {username}"
                id="transferred_to"
                values={
                  Object {
                    "username": <Link
                      replace={false}
                      to="/@hellosteem"
                    >
                      hellosteem
                    </Link>,
                  }
                }
              />
              <span
                className="UserWalletTransactions__transfer"
              >
                - 
                100 STEEM
              </span>
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <FormattedRelative
                updateInterval={10000}
                value="0Z"
              />
            </span>
            <span
              className="UserWalletTransactions__memo"
            >
              Test Transfer Transaction
            </span>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="UserWalletTransactions__transaction"
        >
          <div
            className="UserWalletTransactions__avatar"
          >
            <Avatar
              size={40}
              username="hellosteem"
            />
          </div>
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <FormattedMessage
                defaultMessage="Transferred to {username}"
                id="transferred_to"
                values={
                  Object {
                    "username": <Link
                      replace={false}
                      to="/@hellosteem"
                    >
                      hellosteem
                    </Link>,
                  }
                }
              />
              <span
                className="UserWalletTransactions__transfer"
              >
                - 
                100 STEEM
              </span>
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <FormattedRelative
                updateInterval={10000}
                value="0Z"
              />
            </span>
            <span
              className="UserWalletTransactions__memo"
            >
              Test Transfer Transaction
            </span>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TransferTransaction
    amount="100 STEEM"
    memo="Test Transfer Transaction"
    timestamp="0"
    to="hellosteem"
  />,
}
`;
